bigtop_toolchain/bin/puppetize.sh

```diff
@@ -20,51 +20,74 @@ if [ -f /etc/os-release ]; then
         dnf -y install yum-utils
         dnf -y check-update
-        dnf -y install hostname diffutils findutils curl sudo unzip wget puppet procps-ng libxcrypt-compat systemd
+        dnf -y install hostname diffutils findutils curl sudo unzip wget puppet-agent procps-ng libxcrypt-compat systemd
         # On Fedora 31, the puppetlabs-stdlib package provided by the distro installs the module
         # into /usr/share/puppet/modules, but it's not recognized as the default module path.
         # So we install that module in the same way as CentOS 7.
-        puppet module install puppetlabs-stdlib --version 4.12.0
+        /opt/puppetlabs/bin/puppet module install puppetlabs-stdlib 
         ;;
     ubuntu-18.04|ubuntu-20.04)
         apt-get update
-        apt-get -y install wget curl sudo unzip puppet software-properties-common puppet-module-puppetlabs-apt puppet-module-puppetlabs-stdlib systemd-sysv
+        apt-get -y install wget curl sudo unzip software-properties-common systemd-sysv
+        wget -P /tmp ${APT_DOWNLOAD_URL}-release-bionic.deb
+        dpkg -i /tmp/puppet7${NIGHTLY}-release-bionic.deb
+        apt-get update
+        apt-get -y install puppet-agent
+        /opt/puppetlabs/bin/puppet module install puppetlabs-stdlib
+        /opt/puppetlabs/bin/puppet module install puppetlabs-apt 
         ;;
     debian-10*|debian-11*)
         apt-get update
-        apt-get -y install wget curl sudo unzip puppet puppet-module-puppetlabs-apt puppet-module-puppetlabs-stdlib systemd-sysv gnupg procps
+        apt-get -y install wget curl sudo unzip systemd-sysv gnupg procps
+        wget -P /tmp ${APT_DOWNLOAD_URL}-release-buster.deb
+        dpkg -i /tmp/puppet7${NIGHTLY}-release-buster.deb
+        apt-get update
+        apt-get -y install puppet-agent
+        /opt/puppetlabs/bin/puppet module install puppetlabs-stdlib
+        /opt/puppetlabs/bin/puppet module install puppetlabs-apt
         ;;
     centos-7*)
         rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
-        yum updateinfo
-        # BIGTOP-3088: pin puppetlabs-stdlib to 4.12.0 as the one provided by
-        # distro (4.25.0) has conflict with puppet<4. Should be removed once
-        # puppet in distro is updated.
-        yum -y install hostname curl sudo unzip wget puppet
-        puppet module install puppetlabs-stdlib --version 4.12.0
+        rpm -Uvh ${RPM_DOWNLOAD_URL}-release-el-7.noarch.rpm
+        yum update -y
+        yum -y install hostname curl sudo unzip wget puppet-agent
+        # bump puppet to puppet7
+        /opt/puppetlabs/bin/puppet module install puppetlabs-stdlib
         ;;
     centos-8*)
         sed -i -e 's/^\(mirrorlist\)/#\1/' -e 's,^#baseurl=http://mirror.centos.org,baseurl=https://vault.centos.org,' /etc/yum.repos.d/CentOS-Linux-*
         ;&
     rocky-8*)
         rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
```

Comment:
why we need still have this..? we can remove right as puppet is not there in epel.?

Reply:
yes, the puppet is not there in epel now